applications/MappingApplication/custom_utilities/mapper_utilities.h

```diff
@@ -228,6 +228,149 @@ class MapperUtilities
         return NumNodesOrigin / NumNodesDestination;
     }
 
+    // **********************************************************************
+    // Functions related to Bounding Boxes **********************************
+    // **********************************************************************
+    static void ComputeLocalBoundingBox(ModelPart& rModelPart,
+                                        double* pLocalBoundingBox)
+    {
+        // xmax, xmin,  ymax, ymin,  zmax, zmin
+        // loop over local nodes
+        for (auto &r_node : rModelPart.GetCommunicator().LocalMesh().Nodes())
+        {
+            pLocalBoundingBox[0] = std::max(r_node.X(), pLocalBoundingBox[0]);
+            pLocalBoundingBox[1] = std::min(r_node.X(), pLocalBoundingBox[1]);
+            pLocalBoundingBox[2] = std::max(r_node.Y(), pLocalBoundingBox[2]);
+            pLocalBoundingBox[3] = std::min(r_node.Y(), pLocalBoundingBox[3]);
+            pLocalBoundingBox[4] = std::max(r_node.Z(), pLocalBoundingBox[4]);
+            pLocalBoundingBox[5] = std::min(r_node.Z(), pLocalBoundingBox[5]);
+        }
+        // loop over ghost nodes (necessary if conditions have only ghost nodes)
+        for (auto &r_node : rModelPart.GetCommunicator().GhostMesh().Nodes())
```

Comment:
If you are going to check both local and ghost nodes, you can simply iterate over ModelPart.Nodes()